packages/rum-react/src/get-apm-route.js

```diff
@@ -30,10 +30,23 @@ import { getWithTransaction } from './get-with-transaction'
 function getApmRoute(apm) {
   const withTransaction = getWithTransaction(apm)
 
-  return function ApmRoute(props) {
-    const { path, component } = props
-    const apmComponent = withTransaction(path, 'route-change')(component)
-    return <Route {...props} component={apmComponent} />
+  return class ApmRoute extends React.Component {
+    constructor(props) {
+      super(props)
+      const { path, component } = props
+      /**
+       * Having the ApmComponent on state ensures that we capture the SPA
+       * navigation only when route changes/mount and not on re-renders
+       * Ex: updating query params from child components
+       */
+      this.state = {
+        apmComponent: withTransaction(path, 'route-change')(component)
```

Comment:
what happens when path or component are updated?

Reply:
The below test covers the path case https://github.com/elastic/apm-agent-rum-js/pull/748/files#diff-9cf03e1a91b2eaf59d727f74ebc3503fR134

Does it make sense?

Reply:
`path` as a component prop, I mean (same for `component`).

Reply:
Can you give an example? That would mean, you will have different `ApmRoute` components. 

```jsx
<ApmRoute path="/home" component={MainComponent} />
<ApmRoute path="/func" component={FunctionalComponent} />
```

Reply:
Consider:

```
import React from 'react';

function App ( ) {
  const isLoggedIn = useLoggedIn();

  const path = isLoggedIn ? `/u/0` : '/';

  const component = isLoggedIn ? UserHome : AnonHome;

  return (<>
    <AppComponent path={path} component={component}/>
  </>)
}
```

Obviously a contrived example, but I think consumers _can_ use it in this way? Whether it's a good idea is a different discussion 😬

Reply:
Good one. Not sure a super valid usecase for route components as its better to keep them static instead of dynamically changing the path and underlying component during each render. I will check if we can provide a solution for this as well.

Reply:
I was wrong, I had a quick look. at `Switch` and it seems to do something similar. I changed the solution to use `getDerivedStateFromProps` which should fix this case as well. Thanks 👍